ext/random/engine_combinedlcg.c

```diff
@@ -97,10 +97,12 @@ static bool unserialize(php_random_status *status, HashTable *data)
 
 	for (i = 0; i < 2; i++) {
 		t = zend_hash_index_find(data, i);
-		if (!t || Z_TYPE_P(t) != IS_STRING) {
+		if (!t || Z_TYPE_P(t) != IS_STRING || Z_STRLEN_P(t) != 8 /* sizeof(uint32_t) */) {
```

Comment:
Why the literal integer and the comment, while you actually used sizeof() a few lines above (idem in other files)?

Reply:
I assumed that the size changed depending on the environment, but it was `uint_fast32_t`. Fix it `sizeof(uint32_t)`.

Reply:
Note that you'll need to multiply it by `2`, because the hexadecimal encoding requires 2 bytes to represent one byte.

Reply:
Oops, since this is a string, sizeof could not be used

Reply:
Should I do sizeof * 2 and let the compiler optimize for readability rather than using literals?

Reply:
I would say the following would make the most sense:

```
Z_STRLEN_P(t) != (2 * sizeof(*s->state))
// or
Z_STRLEN_P(t) != (2 * sizeof(s->state[i]))
```

That ensures that the types are consistent.

Reply:
I thought it was good, but considering PCG64, it might be better to use `(2 * sizeof(uintXX_t)` as usual

Reply:
Choose whatever you are comfortable with.